src/coreclr/vm/callingconvention.h

```diff
@@ -789,6 +830,123 @@ class ArgIteratorTemplate : public ARGITERATOR_BASE
     }
 #endif // TARGET_AMD64
 
+#ifdef TARGET_LOONGARCH64
+    // Get layout information for the argument that the ArgIterator is currently visiting.
+    void GetArgLoc(int argOffset, ArgLocDesc *pLoc)
+    {
+        LIMITED_METHOD_CONTRACT;
+
+        pLoc->Init();
+
+        if (m_hasArgLocDescForStructInRegs)
+        {
+            *pLoc = m_argLocDescForStructInRegs;
+            return;
+        }
+
+        if (TransitionBlock::IsFloatArgumentRegisterOffset(argOffset))
+        {
+            // Dividing by 8 as size of each register in FloatArgumentRegisters is 8 bytes.
+            // TODO: support SIMD.
+            pLoc->m_idxFloatReg = (argOffset - TransitionBlock::GetOffsetOfFloatArgumentRegisters()) / 8;
+
+            if (!m_argTypeHandle.IsNull() && m_argTypeHandle.IsHFA())
+            {
+                _ASSERTE(!"unimplemented on LOONGARCH yet! --HFA");
+                //CorElementType type = m_argTypeHandle.GetHFAType();
+                //pLoc->setHFAFieldSize(type);
+                //pLoc->m_cFloatReg = GetArgSize()/pLoc->m_hfaFieldSize;
+
+            }
+            else
+            {
+                pLoc->m_cFloatReg = 1;
+            }
+            return;
+        }
+
+        int cSlots = (GetArgSize() + 7)/ 8;
+
+        // Composites greater than 16bytes are passed by reference
+        if (GetArgType() == ELEMENT_TYPE_VALUETYPE && GetArgSize() > ENREGISTERED_PARAMTYPE_MAXSIZE)
+        {
+            cSlots = 1;
+        }
+
+        if (!TransitionBlock::IsStackArgumentOffset(argOffset))
+        {
+            pLoc->m_idxGenReg = TransitionBlock::GetArgumentIndexFromOffset(argOffset);
+            pLoc->m_cGenReg = cSlots;
+            {
+                if (GetArgType() == ELEMENT_TYPE_R4)
+                {
+                    pLoc->m_flag = 5;
+                    return;//float by integer-reg.
+                }
+                else if (GetArgType() == ELEMENT_TYPE_R8)
+                {
+                    pLoc->m_flag = 6;
+                    return;//float by integer-reg.
+                }
+                else if (!m_flags)
+                {
+                    return;//no float
+                }
+
+                if ((1 == m_flags) || (2 == m_flags))
+                {
+                    pLoc->m_flag = 7;//first double.
+                    return;
+                }
+
+                if ((cSlots == 1) && (m_flags == 0x5))
+                {
+                    pLoc->m_flag = 2;//two float;    ? not case ?
+                    assert(!"----------two float;    ? not case ?------------");
+                }
+                else if ((cSlots == 1) && (m_flags & 0x1))
+                    pLoc->m_flag = 1;//first float;
```

Comment:
Could you please rename the `m_flag` to something more descriptive? It would also be nice to use an enum for the values of m_flag.

The same for `m_flags`.

Reply:
> Could you please rename the `m_flag` to something more descriptive? It would also be nice to use an enum for the values of m_flag.
> 
> The same for `m_flags`.

Thanks.
How about renaming it by `m_structFloatFlags` as it means for the struct-arg that includes float fields and can be passed by one or two float register(s) ?
For this case, we have to descript the float field's info by this vaues.

Reply:
> Could you please rename the `m_flag` to something more descriptive? It would also be nice to use an enum for the values of m_flag.
> 
> The same for `m_flags`.

Thanks.
How about renaming it by `m_structFloatFlags` as it means for the struct-arg that includes float fields and can be passed by one or two float register(s) ?
For this case, we have to descript the float field's info by this vaues.

Reply:
> Could you please rename the `m_flag` to something more descriptive? It would also be nice to use an enum for the values of m_flag.
> 
> The same for `m_flags`.

Thanks.
How about renaming it by `m_structFloatFlags` as it means for the struct-arg that includes float fields and can be passed by one or two float register(s) ?
For this case, we have to descript the float field's info by this vaues.

Reply:
The suggested name sounds good to me.

Reply:
Thanks